gnc/navigator_controller/src/mrac_controller.py

```diff
@@ -336,7 +377,12 @@ def increment_reference(self):
 
         # Use model drag to find drag force on virtual boat
         twist_body = R_ref.T.dot(np.concatenate((self.v_ref, [self.w_ref])))
-        D_body = np.array([p if v > 0 else n for p, n, v in zip(self.D_body_positive, self.D_body_negative, twist_body)])
```

Comment:
I changed its structure because it always left D_body with size (1,) for some reason instead of filling it properly like the normal for loop I replaced it with does